cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -592,6 +592,39 @@ VersionedItem LocalVersionedEngine::write_versioned_metadata_internal(
     }
 }
 
+std::vector<VersionedItem> LocalVersionedEngine::batch_write_versioned_metadata_internal(
+    std::vector<StreamId> stream_ids,
+    bool prune_previous_versions,
+    std::vector<arcticdb::proto::descriptors::UserDefinedMetadata>&& user_meta_proto_vec) {
+    auto stream_update_info_vector_fut = batch_get_latest_undeleted_version_and_next_version_id_async(store(),
+                                                                                                      version_map(),
+                                                                                                      stream_ids);
+    std::vector<folly::Future<VersionedItem>> output_vect_fut;
+    for(const auto& stream_id : folly::enumerate(stream_ids)) {
+        output_vect_fut.push_back(
+            std::move(stream_update_info_vector_fut[stream_id.index])
+            .thenValue([this, &stream_id, user_meta_proto = std::move(user_meta_proto_vec[stream_id.index])](auto&& update_info) mutable {
+                util::check(update_info.previous_index_key_.has_value(),
```

Comment:
OK, this answers my previous question about what the behaviour is if the symbol doesn't exist. `folly::collect` short-circuits on the first exception, so the user will not know which symbols (if any) had metadata written. Would be better to log and return `None` rather than a versioned item in this case.
Unfortunately we will probably need to maintain backwards compatibility with the `NativeVersionStore API`. See my PR for an example of how to do this - https://github.com/man-group/ArcticDB/pull/544

Reply:
Also, always use `internal::check` in preference to `util::check` now